build/test-aot.ps1

```diff
@@ -0,0 +1,57 @@
+param([string]$targetNetFramework)
+
+$projectName='Microsoft.IdentityModel.AotCompatibility.TestApp'
+$rootDirectory = Split-Path $PSScriptRoot -Parent
+$publishOutput = dotnet publish $rootDirectory/test/$projectName/$projectName.csproj --self-contained -nodeReuse:false /p:UseSharedCompilation=false
+
+$actualWarningCount = 0
+
+foreach ($line in $($publishOutput -split "`r`n"))
+{
+    if (($line -like "*analysis warning IL*") -or ($line -like "*analysis error IL*"))
+    {
+        Write-Host $line
+        $actualWarningCount += 1
+    }
+}
+
+Write-Host "Actual warning count is: ", $actualWarningCount
+$expectedWarningCount = 0
+
+if ($LastExitCode -ne 0)
+{
+    Write-Host "There was an error while publishing AotCompatibility Test App. LastExitCode is:", $LastExitCode
+    Write-Host $publishOutput
+}
+
+#$runtime = if ($IsWindows) {  "win-x64" } elseif ($IsMacOS) { "macos-x64"} else {"linux-x64"}
```

Comment:
Can we keep the script able to run on all 3 OSes? That way devs can run the script locally without having to move to a specific OS.

Reply:
I have restored the checks.